contracts/contracts/GrantRegistry.sol

```diff
@@ -1,135 +1,132 @@
-  // while its in config mode, then switch to active. Once active, the owner can only
-  // close the registry (have a conversation around this approach)
-  constructor(
-    address _owner,
-    bytes32 _grantRegistryId,
-    string memory _metaPtr
-  ) {
-    owner = _owner;
-    //state = State.CONFIG;
-    grantRegistryId = _grantRegistryId;
-    metaPtr = _metaPtr;
-  }
-
-  // State
-  enum State {
-    CONFIG,
-    ACTIVE,
-    CLOSED
-  }
-  //State public state;
-
-  // Grant Pointer Object
-  // Mapping from GrantID to set of Owner Address, Payout Address, Metadata Pointer, Replaces Grant (optional)
-  // TODO: Implement safemath for avoiding overflows with ints
-  uint256 public grantCount = 0;
-  mapping(bytes32 => Grant) public grants;
-  bytes32[] public grantIds;
+  // --- Data ---
+  /// @notice Number of grants stored in this registry
+  uint96 public grantCount;
 
+  /// @notice Grant object
   struct Grant {
-    bytes32 id;
-    address owner;
-    address payout;
-    string metaPtr;
-    State state;
-    bytes32 replaces;
+    uint96 id; // grant ID, as uint96 to pack into same slot as owner (this implies a max of 2^96-1 = 7.9e28 grants)
+    address owner; // grant owner (has permissions to modify grant information)
+    address payee; // address that receives funds donated to this grant
+    string metaPtr; // URL pointing to grant metadata (for off-chain use)
   }
 
-  function activate(bytes32 _id) external onlyOwner {
-    grants[_id].state = State.ACTIVE;
-  }
+  /// @notice Mapping from Grant ID to grant data
+  mapping(uint96 => Grant) public grants; // TODO use an array instead with ID to index it? Which is better? Will array copy full array to memory in `getAllGrants`?
 
-  function closed(bytes32 _id) public onlyOwner {
-    grants[_id].state = State.CLOSED;
-  }
+  // --- Events ---
+  /// @notice Emitted when a new grant is created
+  event GrantCreated(uint96 indexed id, address indexed owner, address indexed payee, string metaPtr);
 
-  function isConfig(bytes32 _id) public view returns (bool) {
-    return grants[_id].state == State.CONFIG;
-  }
+  /// @notice Emitted when a grant's owner is changed
+  event GrantUpdated(uint96 indexed id, address indexed owner, address indexed payee, string metaPtr);
 
-  function isActive(bytes32 _id) public view returns (bool) {
-    return grants[_id].state == State.ACTIVE;
-  }
-
-  function isClosed(bytes32 _id) public view returns (bool) {
-    return grants[_id].state == State.CLOSED;
+  // --- Core methods ---
+  /**
+   * @notice Create a new grant in the registry
+   * @param _owner Grant owner (has permissions to modify grant information)
+   * @param _payee Address that receives funds donated to this grant
+   * @param _metaPtr URL pointing to grant metadata (for off-chain use)
+   */
+  function createGrant(
+    address _owner,
+    address _payee,
+    string memory _metaPtr
```

Comment:
noob question !
Do we need to worry about grants sharing same _metaPtr or nah it doesn't matter cause the entity would take care of it while creation ?

Reply:
Ideally we don't want two grants sharing the same `metaPtr`, because why would two grants have the same metadata, but I don't think this is worth enforcing on-chain. Probably fine to leave this to the matching eligibility layer, but again tagging @apbendi and @phutchins for input

Reply:
Agree on your approach here @mds1.